cache_flat_mutation_reader.hh

```diff
@@ -642,10 +657,38 @@ void cache_flat_mutation_reader::copy_from_cache_to_buffer() {
     }, _read_context.is_reversed()) == stop_iteration::no) {
         return;
     }
-    // We add the row to the buffer even when it's full.
-    // This simplifies the code. For more info see #3139.
+
     if (_next_row_in_range) {
-        add_to_buffer(_next_row);
+        bool remove_row = false;
+
+        if (_state == state::reading_from_cache && !_next_row.dummy()) { // maybe we'd better use at_a_row() instead of dummy() here?
+            deletable_row& row = _next_row.latest_row();
+            tombstone base_tombstone = _snp->version()->partition().range_tombstone_for_row(*_schema, _next_row.key());
```

Comment:
That will be very expensive, the price is too high to pay per-row.

We should either wait for https://github.com/scylladb/scylladb/pull/12048, or make this compaction weaker by not considering range tombstones. 

You could ignore base_tombstone in is_live() since we're not really interested

Reply:
I agree, that really could be expensive but don't know better solution in current implementation.
Do you think we can just ignore range tombstones?

Reply:
I recommend to rebase your patches on top of #12048, it will allow everything to be considered.

Reply:
Thanks, I did. However I could apply my changes (based on #12048) only when it will be finished